File: backend/app/schemas/task.py
from datetime import datetime
from enum import Enum
from typing import Optional
from uuid import UUID

from pydantic import BaseModel


class TaskPriority(str, Enum):
    LOW = "Low"
    MEDIUM = "Medium"
    HIGH = "High"


class TaskStatus(str, Enum):
    TODO = "Todo"
    IN_PROGRESS = "In Progress"
    COMPLETED = "Completed"


class TaskCreate(BaseModel):
    project_id: UUID
    title: str
    description: Optional[str] = None
    priority: TaskPriority = TaskPriority.MEDIUM
    due_date: Optional[datetime] = None


class TaskUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    status: Optional[TaskStatus] = None
    priority: Optional[TaskPriority] = None
    due_date: Optional[datetime] = None


class TaskResponse(BaseModel):
    id: UUID
    project_id: UUID
    title: str
    description: Optional[str]
    status: TaskStatus
    priority: TaskPriority
    due_date: Optional[datetime]
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/lib/validators/task.ts
import { z } from "zod";

export const taskSchema = z.object({
  project_id: z.string().min(1),

  title: z
    .string()
    .min(3, "Title must be at least 3 characters"),

  description: z.string().optional(),

  priority: z.enum([
    "Low",
    "Medium",
    "High",
  ]),

  status: z.enum([
    "Todo",
    "In Progress",
    "Completed",
  ]),
});

export type TaskFormData =
  z.infer<typeof taskSchema>;